docs/advanced/additional-requests.rst

```diff
@@ -0,0 +1,802 @@
+            return item
+
+        @staticmethod
+        def parse_related_product_ids(response_page) -> List[str]:
+            return response_page.css("#main .related-products ::attr(product-id)").getall()
+
+Here's the key takeaway in this example:
+
+    * Similar to :class:`~.HttpClient`'s :meth:`~.HttpClient.get` method,
+      a :meth:`~.HttpClient.post` method is also available that's
+      typically used to submit forms.
+
+.. _`http-batch-request-example`:
+
+Batch requests
+--------------
+
+We can also choose to process requests by **batch** instead of sequentially or 
+one by one. The :meth:`~.HttpClient.batch_requests` method can be used for this
+which accepts an arbitrary number of :class:`~.HttpRequest` instances.
+
+Let's modify the example in the previous section to see how it can be done.
+
+The difference for this code example from the previous section is that we're
+increasing the pagination from only the **2nd page** into the **10th page**.
+Instead of calling a single :meth:`~.HttpClient.post` method, we're creating a
+list of :class:`~.HttpRequest` to be executed in batch using the
+:meth:`~.HttpClient.batch_requests` method.
+
+.. code-block:: python
+
+    from typing import List
+
+    import attrs
+    import web_poet
+
+
+    @attrs.define
+    class ProductPage(web_poet.ItemWebPage):
+        http_client: web_poet.HttpClient
+
+        default_pagination_limit = 10
+
+        async def to_item(self):
+            item = {
+                "url": self.url,
+                "name": self.css("#main h3.name ::text").get(),
+                "product_id": self.css("#product ::attr(product-id)").get(),
+                "related_product_ids": self.parse_related_product_ids(self),
+            }
+
+            requests: List[web_poet.HttpRequest] = [
+                self.create_request(page_num=page_num)
+                for page_num in range(2, self.default_pagination_limit)
+            ]
+            responses: List[web_poet.HttpResponse] = await self.http_client.batch_requests(*requests)
+            related_product_ids = [
+                id_
+                for response in responses
+                for product_ids in self.parse_related_product_ids(response)
+                for id_ in product_ids:
+            ]
+
+            item["related_product_ids"].extend(related_product_ids)
+            return item
+
+        def create_request(self, page_num=2):
+            # Simulates "scrolling" through a carousel that loads related product items
+            return web_poet.HttpRequest(
+                url="https://www.api.example.com/product-pagination/",
+                method="POST",
+                headers={
+                    "Content-Type": "application/json;charset=UTF-8"
+                },
+                body=json.dumps(
+                    {
+                        "Page": page_num,
+                        "ProductID": item["product_id"],
```

Comment:
item is not defined

Reply:
Nice catch! Updated in https://github.com/scrapinghub/web-poet/pull/22/commits/80f95a8f9a882e4fb3579f546e10bd54801z6e172.